src/terminal/terminal-handlers.c

```diff
@@ -835,6 +835,24 @@ int guac_terminal_csi(guac_terminal* term, unsigned char c) {
 
                 break;
 
+            /* Scroll Up by amount */
+            case 'S':
+
+                /* Multiple linefeed */
+                for (int i = 0; i < argv[0]; i++)
+                    guac_terminal_linefeed(term, false);
```

Comment:
One other question - above, in the sections for inserting/deleting lines, there are functions `guac_terminal_scroll_up` and `guac_terminal_scroll_down` - is there a reason not to use these, versus using `for` loops you have here?

Reply:
Yes, we can use these functions directly instead of doing a for loop.
I changed and tested again: the result is the same.